velox/connectors/hive/HiveConnectorUtil.cpp

```diff
@@ -681,7 +687,8 @@ bool testFilters(
     const std::unordered_map<std::string, std::optional<std::string>>&
         partitionKeys,
     const std::unordered_map<std::string, std::shared_ptr<HiveColumnHandle>>&
-        partitionKeysHandle) {
+        partitionKeysHandle,
+    bool isPartitionValueDaysSinceEpoch) {
```

Comment:
This cannot be a single value, what if one partition key is encoded as int and another encoded as ISO string?  This has to be a property of the column handle.

Reply:
@Yuhta Thanks for the review. The DATE partition value is int(daysSinceEpoch) for Iceberg and ISO string(YYYY-MM-DD) for Hive. This flag will be true if we are reading Iceberg splits. No other combination exists.

Reply:
I have tested with filters on non-partitioned DATE column too, it works fine.

Reply:
It's a very restricted design that only works because Iceberg use int for all dates.  The config should be per column by nature.

Reply:
Makes sense to make this a column-level property.

Reply:
Yeah. Shall I change the Presto's ColumnHandle classes, the prestocpp protocol and Velox's IcebergColumnHandle and HiveColumnHandle or should we for now change only Velox's HiveColumnHandle when converting from IcebergColumnHandle?

Reply:
@Yuhta @majetideepak

Reply:
@nmahadevuni It is enough to add to Velox's HiveColumnHandle when converting from IcebergColumnHandle. This happens in Prestissimo anyway. We can improve the protocol there if needed later.